### DISCUSSION TITLE
Clarification Regarding AutoGen Forks

### DISCUSSION THREAD

**Discussion by jackgerrits** [microsoft/autogen#4217] [Announcements]
[_Originally posted on X_](https://x.com/pyautogen/status/1857264760951296210)

To all our collaborators and community members, 

There have been a number of asks to clarify and distinguish between various AutoGen forks and clones and the current pyautogen package.

Here are some important things to know:
- The official GitHub repository for the AutoGen project is still [github.com/microsoft/autogen](https://github.com/microsoft/autogen/).
- We are actively working on and maintaining [AutoGen 0.2](https://microsoft.github.io/autogen/0.2/docs/Getting-Started) and have recently introduced new innovations to [AutoGen 0.4](https://microsoft.github.io/autogen/dev/) (AutoGen-Core, Autogen-AgentChat), as well as  [AutoGen-Studio](https://github.com/microsoft/autogen/tree/main/python/packages/autogen-studio) and [Magentic-One](https://aka.ms/magentic-one) as part of the project’s roadmap.
- Development will continue in the open-source under the MIT license. We are grateful to all past contributors and welcome anyone to use or contribute to the project.
- We are currently restructuring our python packages, see details [here](https://aka.ms/autogen/packages). Unfortunately, admin access to pyautogen is currently blocked for us. Please be aware that the current version of pyautogen is not affiliated with the [microsoft/autogen](https://github.com/microsoft/autogen) project.
- We are unable to communicate on the original AutoGen Discord server,  since many of our members no longer have viewing or posting permissions there. For inquiries and discussions, please reach out to us on [GitHub Discussions](https://github.com/microsoft/autogen/discussions), via email at [autogen@microsoft.com](mailto:mautogen@microsoft.com) or join our community office hours on Wednesdays 10am PST/1pm EST/7pm CET at –  [Teams meeting link](https://teams.microsoft.com/meet/222836470934?p=3bs3pOaAtlftpTiCRh). Update (12/13): we've launched a new [discord server](https://aka.ms/autogen-discord).

Since its creation, AutoGen has been a highly collaborative project and most of the team behind AutoGen from day one continues to be deeply involved today. While we want to clarify that no one at Microsoft is involved with any forks or clones of this project, we recognize that these variations reflect the diverse interests and evolving business endeavors in the community. Our team continues to remain dedicated to working with the amazing community that has helped shape this project.

Thank you for your continued support and stay tuned for new releases coming soon.

**Comment by fniedtner** [comment]
@qingyun-wu and @sonichi, we appreciate your contributions to AutoGen’s success.

AutoGen’s success, evolution, and maintenance since its beginning has been shaped by efforts of many with hundreds of contributors to the code base and more than 20 Microsoft folks dedicating all or some of their time to it over the last year.

We are committed to developing and supporting the project in open-source and keep it open to all contributors. We recognize there has been some unfortunate confusion about forks, changes in access to pypi packages, and blocked permission for participation on Discord for many of our team members. See [Clarification Regarding AutoGen Forks · microsoft/autogen · Discussion #4217](https://github.com/microsoft/autogen/discussions/4217) .

The AutoGen team remains committed to the mission of AutoGen and looks forward to continuing our collaboration with the open-source community. We are excited to very soon release AutoGen 0.4, and new innovations with Magentic-1, AutoGen Studio and more. We remain excited and committed to building game-changing technology for the future of agentic technology with the open-source community.

On behalf of the AutoGen contributors at Microsoft

**Reply by xianminx** [reply]
I first learned about AutoGen through Andrew Ng's Coursera course taught by Chi Wang. I’ve noticed that **ag2** has now become the primary fork of Microsoft’s `autogen`, with Chi Wang as its core contributor. According to his LinkedIn profile, Chi Wang now works at Google DeepMind and previously spent over a decade at Microsoft. He also claims to be the founder of AutoGen.

One point of confusion is the licensing: **ag2** is now released under the Apache 2.0 license, whereas the original AutoGen remains under the MIT license. 

As users and developers of AutoGen, this raises a key question: which library should we follow going forward? The divergence in both development focus and licensing makes it unclear which version will best serve our needs in the long term.

**Reply by ekzhu** [reply]
Would you like to share your scenarios and your needs for an agent framework, in a separate discussion thread?

Many developers of the original AutoGen version are still here, listening, curious and constantly pushing the boundaries on both multi-agent capabilities and foundational AI system. We don't claim titles such "founder" or "creator" that exclude others in a collaborative project.

**Reply by fniedtner** [reply]
For additional clarity I would like to emphasize what @ekzhu said, Microsoft contributors continue to evolve AutoGen - as many of us have done since the project's early days. Adding that AG 0.4 will be released soon, it's a re-design that that we believe to be a big step forwards.
